Modules/_testcapimodule.c

```diff
@@ -5652,6 +5652,42 @@ test_macros(PyObject *self, PyObject *Py_UNUSED(args))
     Py_RETURN_NONE;
 }
 
+static PyObject *
+function_get_code(PyObject *self, PyObject *func)
+{
+    PyObject *code = PyFunction_GetCode(func);
+    if (code != NULL) {
+        Py_INCREF(code);
```

Comment:
Should this `Py_INCREF` be there? 🤔 

Without it I have this error:

```
test_function_get_module (test.test_capi.CAPITest.test_function_get_module) ... ok

----------------------------------------------------------------------
Ran 1 test in 0.001s

OK

== Tests result: SUCCESS ==

1 test OK.

Total duration: 242 ms
Tests result: SUCCESS
./Include/object.h:625: _Py_NegativeRefcount: Assertion failed: object has negative ref count
<object at 0x1082424d0 is freed>
Fatal Python error: _PyObject_AssertFailed: _PyObject_AssertFailed
Python runtime state: finalizing (tstate=0x000000010759c6d0)

Current thread 0x000000011215d5c0 (most recent call first):
  <no Python frame>
[1]    92538 abort      ./python.exe -m test -v test_capi -m test_function_get_module
```

Reply:
Cc: @markshannon as it pertains to code objects.

Reply:
I think this is correct, `PyFunction_GetCode` is documented as returning a borrowed ref: https://docs.python.org/3/c-api/function.html#c.PyFunction_GetCode